js/app/modules/contentGroup/contentGroup.js

```diff
@@ -166,9 +166,14 @@ const ContentGroup = new Module.Class({
 
     _on_models_changed: function () {
         let models = this._selection.get_models();
-        if (models.length === 0 && this._no_results) {
+        this.visible = true;
+        if (models.length === 0) {
             this._arrangement.set_models([]);
-            this._stack.visible_child_name = NO_RESULTS_PAGE_NAME;
+            if (this._no_results) {
```

Comment:
Maybe I'll understand with more context; which content group is this for, and when is it supposed to show nothing? I'm having trouble imagining a use case for this that isn't bad UX.